themes/default/upload_image.tpl

```diff
@@ -7,134 +7,286 @@
+  margin-block: 0.5em;
+}
+h1 {
+  font-size: 1em;
+  font-weight: bold;
+}
+.caution,
+.ok {
+  margin-block: 0.5em;
+  padding: 0 0 0 24px;
+  background-repeat:no-repeat;
+  background-position: 2px 3px;
+}
+.caution h2,
+.ok h2 {
+  font-size: 1em;
+  margin: 0 0 0.5em 0;
+}
+.caution {
+  color: #cc0000;
+  background-image:url({/literal}{$THEMES_DIR}/{$settings.theme}{literal}/images/caution.png);
+}
+.ok {
+  color:green;
+  background-image:url({/literal}{$THEMES_DIR}/{$settings.theme}{literal}/images/tick.png);
+}
+.deletelink {
+  padding-left:13px;
+  background:url({/literal}{$THEMES_DIR}/{$settings.theme}{literal}/images/bg_sprite_3.png) no-repeat 1px -45px;
+}
+.insert-desc,
+.deletelink,
+.small {
+  font-size: 0.82em;
+}
+code {
+  font-family:"courier new", courier, monospace;
+  color:#000080;
+}
+a {
+  color:#00c;
+  text-decoration: none;
+}
+a:focus,
+a:hover {
+  color:#00f;
+  text-decoration: underline dotted 9% #45f;
+}
+a:active {
+  color:#f00;
+  text-decoration: underline solid 7% #d00;
+}
+form > div:not(:last-child) {
+  margin-block-end: .75em;
+}
+ul {
+  list-style:none;
+  margin-block:0.5em;
+  padding: 0;
+}
+#imgtab {
+  display:flex;
+  flex-direction:column;
+  gap:0.75em;
+}
+#del-upload-form,
+#imgtab li {
+  text-align:center;
+}
+li > *:last-child {
+  align-content:center;
+}
+img {
+  border:none;
+  display: block;
+}
+#imgtab img {
+  max-width: 100%;
+  margin-inline: auto;
+}
+#imgtab:not(.shrinked) img {
+  cursor: pointer;
+}
+#imgtab.shrinked img {
+  max-width: 50%;
+  height: auto;
+}
+.buttonbar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 0.3em;
+}
+button {
+  cursor: pointer;
+}
+#imgtab li button:has(> img) {
+  background: transparent;
+  border: none;
+  padding: 0;
+}
 -->
 {/literal}
   </style>
-  <script type="text/javascript">{literal}/* <![CDATA[ */
-function insertCode(image_url) {
- 	if (opener && opener.mlfBBCodeButton) {
-		var bbcodeButton = opener.mlfBBCodeButton;
-		if (!bbcodeButton.canInsert()) 
-			return;
-		var buttonGroup = bbcodeButton.getButtonGroup();	
-		var txtarea = buttonGroup.getTextArea();
-		txtarea.insertTextRange( txtarea.getSelection() + "[img]" + image_url + "[/img]" );
-	}
-	//self.close();
-}
-/* ]]> */{/literal}</script>
-{if $browse_images}
-  <script type="text/javascript">{literal}/* <![CDATA[ */
-function getMaxWidth()
- {
-  if(document.getElementById('imgtab'))
-   {
-    var maxWidth = document.getElementById('imgtab').offsetWidth-20;
-    var obj=getElementsByClassName('browse');
-    for(i=0;i<obj.length;i++)
-     {
-      obj[i].style.maxWidth=maxWidth+'px';
-     }
-   }
- }
-window.onload = getMaxWidth;
-window.onresize = getMaxWidth;
-/* ]]> */{/literal}</script>
-{/if}
+  <script>{literal}
+/**
+ * function for inserting uploaded images into
+ * a posting from the uploaded images gallery
+ */
+function insertCode() {
+  const clickedButton = event.target.closest('button');
+  if (clickedButton === null) return false;
```

Comment:
Personally, I don't like this all-in-a-single-line style and and I don't think it's a good practice because it makes the code hard to read. There are dozens of places in MLF that I could swear at every time. Anyway, it works as accepted. I'm able to upload images, to delete images and to insert images. SVG is still not supported but this is not a template issue. It is interesting that SVGs are shown in the admin area, isn't it?

Reply:
Which line you are complaining about? Is it 155 or 156?

Reply:
I know that we have different preferences about this. :-)

I will change the if-then one liner.

Reply:
I reformatted the one liners to return on invalid states to your preferred style. Longer lines with concatenations of anything, that prevents additional variables to only store one-time interim values for the use in the next line remains.